ros2launch/ros2launch/command/launch.py

```diff
@@ -109,33 +110,15 @@ def main(self, *, parser, args):
         else:
             raise RuntimeError('unexpected mode')
         launch_arguments.extend(args.launch_arguments)
-        try:
-            if args.show_all_subprocesses_output:
-                os.environ['OVERRIDE_LAUNCH_PROCESS_OUTPUT'] = 'both'
-            if args.print:
-                return print_a_python_launch_file(python_launch_file_path=path)
-            elif args.show_args:
-                return print_arguments_of_python_launch_file(python_launch_file_path=path)
-            else:
-                return launch_a_python_launch_file(
-                    python_launch_file_path=path,
-                    launch_file_arguments=launch_arguments,
-                    debug=args.debug
-                )
-        except SyntaxError:
-            print("""
-Notice: SyntaxError (or related errors) can occur if your launch file ('{}') is not a Python file.
-""".format(path), file=sys.stderr)
-            raise  # raise so the user can see the traceback in useful SyntaxError's
-        except ValueError as exc:
-            print("""
-Notice: ValueError can occur if your launch file ('{}') is a binary file and not a Python file.
-""".format(path), file=sys.stderr)
-            raise RuntimeError('ValueError: {}'.format(str(exc)))
-        except InvalidPythonLaunchFileError as exc:
-            # TODO(wjwwood): refactor this after we deprecate and then remove the old launch
-            print("""
-Notice: Your launch file may have been designed to be used with an older version of ROS 2.
-Or that the file you specified is Python code, but not a launch file.
-""", file=sys.stderr)
-            raise RuntimeError('InvalidPythonLaunchFileError: {}'.format(str(exc)))
```

Comment:
Where did all of these special error handlers go?

Reply:
I guess handled in part by https://github.com/ros2/launch/pull/271 and in part by only trying to interpret a launch file as a Python script if it ends in `.py`?

Reply:
After https://github.com/ros2/launch/pull/271, it works like this:
- Extension recognized: The "correct" loader (based on the extension) is tried first. If it fails, all the other loaders are also tried. If all of them fail, the exception of the "correct" loader is shown.
- Without extensions/not recognized extension: All loaders are tried. In case of failure a generic message is shown.

I should maybe move this special error handlers to `get_launch_description_from_python_launch_file`. I will do that.

The error raised by `get_launch_description_from_any_launch_file` is always `InvalidLaunchFileError`. The error information of the previously raised error is wrapped in it.

Reply:
Sounds good to me, thanks for explaining.